pkg/abi/abi.go

```diff
@@ -315,6 +315,56 @@ func (a ABI) Errors() map[string]*Entry {
 	return m
 }
 
+// Returns the components value from the parsed error
+func (a ABI) ParseError(revertData []byte) (*Entry, *ComponentValue, bool) {
+	return a.ParseErrorCtx(context.Background(), revertData)
+}
+
+// Returns the components value from the parsed error
+func (a ABI) ParseErrorCtx(ctx context.Context, revertData []byte) (*Entry, *ComponentValue, bool) {
+	// Always include the default error
+	a = append(ABI{
+		{Type: Error, Name: "Error", Inputs: ParameterArray{{Name: "reason", Type: "string"}}},
+	}, a...)
+	for _, e := range a {
+		if e.Type == Error {
+			if cv, err := e.DecodeCallDataCtx(ctx, revertData); err == nil {
+				return e, cv, true
+			}
+		}
+	}
+	return nil, nil, false
+}
+
+func (a ABI) ErrorString(revertData []byte) (string, bool) {
```

Comment:
I wonder if the string should have an optional details to give you the error string and the values passed in?

Reply:
Can't understand this comment. Seems like `ParseError` is designed to give you the component values of the decoded error, and `ErrorString` is a shortcut to go straight from bytes to a friendly string. Feels like both options are covered here, unless I'm missing something.

Reply:
Yeah, that is fine - I was thinking you could combine an error such as `Error("Not enough Ether provided.")` with the values passed into the call but that will be responsibility of the caller